parcels/fieldset.py

```diff
@@ -253,7 +252,9 @@ def from_netcdf(cls, filenames, variables, dimensions, indices=None,
             for procvar, _ in fields.items():
                 procdims = dimensions[procvar] if procvar in dimensions else dimensions
                 procinds = indices[procvar] if (indices and procvar in indices) else indices
-                if procdims == dims and procinds == inds:
+                procpaths = filenames[procvar] if type(filenames) is dict and procvar in filenames else filenames
```

Comment:
I try to use `isinstance(filenames, dict)`
except if I check for two types, then I use `type(var) in [type1, type2]`

Do you know if there is a preferred option between isinstance and type ?